System/Random.hs

```diff
@@ -430,22 +433,55 @@ mkStdGen32 sMaybeNegative = StdGen (s1+1) (s2+1)
 createStdGen :: Integer -> StdGen
 createStdGen s = mkStdGen32 $ fromIntegral s
 
-{- |
-With a source of random number supply in hand, the 'Random' class allows the
-programmer to extract random values of a variety of types.
 
-Minimal complete definition: 'randomR' and 'random'.
-
--}
+-- $uniform
+--
+-- Type classes 'Uniform' and 'UniformRange' are unrelated type
+-- classes. This is deliberate decision because there're types which
+-- could have instance for one type class but not the other.
+--
+-- For example: 'Integer', 'Float', 'Double' have instance for
+-- @UniformRange@ but there's no way to define @Uniform@.
+--
+-- Conversely there're types where @Uniform@ instance is possible
+-- while @UniformRange@ is not. One example is:
+--
+-- > instance (Uniform a, Uniform b) => Uniform (a,b)
+--
+-- This instance is very straightforward. @UniformRange@ isn't very
+-- sensible even if try to define one compatible with 'Ord' we'll run
+-- into problem that we don't know how many values inhabit @b@
+-- therefore we don't even know how many values we need to generate.
```

Comment:
To be clear, is what you're saying here that

```haskell
instance (UniformRange a, UniformRange b) => UniformRange (a, b)
```

is not sensible? I think that "@UniformRange@ isn't very sensible" is quite ambiguous.